codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/SymbolVisitor.kt

```diff
@@ -393,6 +396,27 @@ fun Symbol.Builder.rustType(rustType: RustType): Symbol.Builder = this.putProper
 fun Symbol.Builder.module(module: RustModule.LeafModule): Symbol.Builder = this.putProperty(RUST_MODULE_KEY, module)
 fun Symbol.module(): RustModule.LeafModule = this.expectProperty(RUST_MODULE_KEY, RustModule.LeafModule::class.java)
 
+/**
+ * Creates a test module for this symbol.
+ * For example if the symbol represents the name for the struct `struct MyStruct { ... }`,
+ * this function will create the following inline module:
+ * ```rust
+ *  #[cfg(test)]
+ *  mod test_my_struct { ... }
+ * ```
+ */
+fun Symbol.testModule(): RustModule.LeafModule {
+    val module = module()
+
+    return RustModule.new(
+        name = "test_${name.toRustName()}",
+        visibility = Visibility.PRIVATE,
+        inline = true,
+        parent = module.parent,
+        additionalAttributes = listOf(Attribute.Cfg("test")),
+    )
+}
```

Comment:
slight preference for adding this on `RustModule` instead—I think it's a little odd to put it inside symbol visitor (since it doesn't really have anything to do with the symbol). Callers would do `symbol.module().testModule()`

Reply:
That's what I initially wanted to do, but in this case all `symbol`s (constrained types) are defined in the same `model.rs` module, and, from my understanding, I can't access the symbol _from_ the module, which means that for all these types the `symbol.module().testModule()` would be the same (say, `mod test_model`) which complicates codegen a bunch.

I guess I could do something like: `symbol.module().testModule(symbol.name)` too.

Let me know what you think.

Reply:
Moved this to `SymbolProvider.testModuleForShape` as we discussed, in f03bf200dd14f91035f19764832eee4a289f716d.